xarray/core/dataarray.py

```diff
@@ -3505,6 +3505,202 @@ def pad(
         )
         return self._from_temp_dataset(ds)
 
+    def idxmin(
+        self,
+        dim: Hashable = None,
+        skipna: bool = None,
+        fill_value: Any = np.NaN,
```

Comment:
Use `fill_value=dtypes.NA` here. Then use the utilities from `dtypes` to replace it on individual `Variable` objects :
https://github.com/pydata/xarray/blob/6378a711d50ba7f1ba9b2a451d4d1f5e1fb37353/xarray/core/variable.py#L1007-L1010

I think this should make it work on arbitrary dtypes, without any need for dtype specific checks elsewhere.

(This handling of `dtypes.NA` could probably be moved into a separate utility functions in `dtypes.py`, if you like)

Reply:
I don't currently do any dtype-specific checks, I am just letting `where` handle that.  But I will use `dtypes.NA`.